### DISCUSSION TITLE
[QST]!NOT_FOUND <InMemoryTableScanExec> cannot run on GPU

### DISCUSSION THREAD

**Discussion by PerkzZheng** [NVIDIA/spark-rapids#5398] [General] [answered]
**What is your question?**
Hi, I got an GPU warning while using Spark 3.0 with rapdis. This warning starts with the operation of dataframe.cache().

`!NOT_FOUND <InMemoryTableScanExec> cannot run on GPU because no GPU enabled version of operator class org.apache.spark.sql.execution.columnar.InMemoryTableScanExec could be found`

Most of my operations were based on Spark SQL. Specially, dataframe.cache() were applied for in-memory acceleration.  My question is that does this warning represent that it can not scan GPU in-memory data or datasets haven't been cached to GPU?

**Comment by revans2** [comment, marked as answer]
We are still working on optimizing cache and persist for the GPU see #11 

Because of how the code works in spark it is going to require changes to spark to be able to even do things like optimizing the compression of the data using the GPU. Currently, this is being worked on as a part of https://github.com/apache/spark/pull/29067.  But even after those changes, it would not be enough to cache the data in GPU memory. We might be able to hack something together to make that work, but likely even further changes would be required to spark and the block manager to be able to do something like that.

**Comment by revans2** [comment]
@PerkzZheng is it Okay if I close this?

**Comment by PerkzZheng** [comment]
@revans2 Sure, thank you for your reply !
